src/record_syscall.cc

```diff
@@ -2196,42 +2197,46 @@ static Switchable prepare_bpf(RecordTask* t,
     }
     case BPF_PROG_QUERY: {
       auto attr_size = t->regs().arg3();
-      auto attrp = syscall_state.reg_parameter(2, attr_size, IN_OUT)
-        .cast<typename Arch::bpf_attr>();
-      auto end_addr = attrp.as_int() + attr_size;
+      auto attr_begin = syscall_state.reg_parameter(2, attr_size, IN_OUT);
+      auto attr_buf = remote_buffer<void>(attr_begin, attr_size);
```

Comment:
How about just using `MemoryRange` instead of `remote_buffer`?

Reply:
I... looked for it and couldn't find it :clown_face: thanks!